pkg/tfbridge/provider.go

```diff
@@ -1433,6 +1444,73 @@ func (p *Provider) Read(ctx context.Context, req *pulumirpc.ReadRequest) (*pulum
 	return &pulumirpc.ReadResponse{}, nil
 }
 
+// processImportValidationErrors runs `Validate` and then processes any resulting validation errors.
+// This only needs to run during an `Import` operation because the values read from the cloud
+// during import will be turned into the inputs in the generated code. Sometimes the cloud returns invalid
+// input values which will cause the next `up` to fail.
+//
+// To try and avoid this failure case, we run the validate functions for the generated inputs and
+// if any fail (and they match our other cases) then we remove those inputs so that the generated
+// program code will not contain the invalid inputs
+//
+// NOTE: we want to (for now at least) scope this as narrowly as possible because we do not want to
+// create a scenario where the resulting inputs create a different error
+// (e.g. removing an invalid required property). In these cases we will just allow the invalid configuration
+// through and the user will have to decide what to do.
+func (p *Provider) processImportValidationErrors(
+	ctx context.Context,
+	urn resource.URN,
+	tfName string,
+	inputs resource.PropertyMap,
+	schema shim.SchemaMap,
+	schemaInfos map[string]*info.Schema,
+) {
+	logger := GetLogger(ctx)
+	tfInputs, _, err := makeTerraformInputsWithOptions(ctx,
+		&PulumiResource{URN: urn, Properties: inputs},
+		p.configValues, inputs, inputs, schema, schemaInfos,
+		makeTerraformInputsOptions{DisableTFDefaults: true, UnknownCollectionsSupported: p.tf.SupportsUnknownCollections()})
+	if err != nil {
+		return
+	}
+
+	rescfg := MakeTerraformConfigFromInputs(ctx, p.tf, tfInputs)
+	_, errs := p.tf.ValidateResource(ctx, tfName, rescfg)
+	for _, e := range errs {
+
+		path, _, _ := parseCheckError(schema, schemaInfos, e)
+
+		// do not process errors on nested types
```

Comment:
There's a few scenarios unblocked by nested types, let's backlog and link the TODO here.

Reply:
I'll just add a comment on https://github.com/pulumi/pulumi-terraform-bridge/issues/2314 that this PR has addressed part of it, but that we still need to address nested types.